### PR TITLE
[TASK-7344] fix: ask for user auth when getting kyc status

### PR DESCRIPTION
**Pull request by jjramirezn**
Check jwt cookie before informing on a kyc status. 

This PR also brings main to develop

## Summary by CodeRabbit

- **New Features**
	- Enhanced user validation and error handling for user status retrieval.
	- Improved error messaging for insufficient balance in link creation.
	- Updated labels for slippage calculations to clarify user understanding.

- **Bug Fixes**
	- Improved error handling for user authentication and KYC data fetching.

- **Style**
	- Renamed buttons from "Return" to "Go Back" across various components for better clarity.
	- Adjusted padding for responsive design in header and dropdown components.

- **Chores**
	- Simplified code structure in navigation components for better maintainability.
	- Removed obsolete user retrieval API endpoint.

### CODE REVIEW

File: src/app/api/bridge/user/new/get-status/route.ts [hunk 1]
```diff
@@ -1,13 +1,31 @@
...
+        })
+        const getUserFromCookieResponse = await getUserFromCookie(getUserFromCookieRequest)
+        if (!getUserFromCookieResponse.ok) {
+            return new NextResponse('Unauthorized', { status: 401 })
+        }
+        const { user } = await getUserFromCookieResponse.json()
+
+        if (userId !== user?.id) {
+            return new NextResponse('Forbidden', { status: 403 })
+        }
```
[the remark this answers is not in the document]
Comment (jjramirezn) [reply src/app/api/bridge/user/new/get-status/route.ts:27]: Already handled

File: src/components/Request/Pay/Views/Initial.view.tsx [hunk 2]
```diff
@@ -422,12 +422,12 @@ export const InitialView = ({
...
                                 <div className="flex w-max flex-row items-center justify-center gap-1">
                                     <Icon name={'money-out'} className="h-4 fill-gray-1" />
-                                    <label className="font-bold">Slippage cost</label>
+                                    <label className="font-bold">Max slippage</label>
                                 </div>
                                 <label className="flex flex-row items-center justify-center gap-1 text-center text-sm font-normal leading-4">
                                     ${calculatedSlippage}
                                     <MoreInfo
-                                        text={`To ensure the request arrives with at least what was asked, we will charge you ${slippagePercentage!.toFixed(2)}% of the requested amount.`}
+                                        text={`${slippagePercentage!.toFixed(2)}% is the maximum slippage set to ensure that the transaction goes through. It is likely to be much lower than the actual slippage`}
```
[the remark this answers is not in the document]
Comment (jjramirezn) [reply src/components/Request/Pay/Views/Initial.view.tsx:430]: not on scope

File: src/app/api/bridge/user/new/get-status/route.ts [hunk 3]
```diff
@@ -3,11 +1,31 @@
...
             throw new Error('BRIDGE_API_KEY is not defined')
         }
+
+        const getUserFromCookieRequest = new NextRequest('/api/peanut/user/get-user-from-cookie', {
+            method: 'GET',
+            headers: {
+                cookie: request.cookies.toString(),
+                ...request.headers,
+            },
+        })
```
[the remark this answers is not in the document]
Comment (jjramirezn) [reply src/app/api/bridge/user/new/get-status/route.ts]: done

File: src/app/api/peanut/user/get-user-from-cookie/route.ts [hunk 4]
```diff
@@ -1,16 +1,15 @@
...
     const cookieStore = cookies()
     const token = cookieStore.get('jwt-token')
 
     if (!token) {
         return new NextResponse('Bad Request: missing required parameters', { status: 400 })
     }
     try {
-        const { protocol, hostname, port } = new URL(request.url)
-        const apiUrl = `${protocol}//${hostname}:${port}/api/peanut/user/get-user`
+        const apiUrl = `${process.env.NEXT_PUBLIC_BASE_URL}/api/peanut/user/get-user`
```
Comment (Hugo0) [inline src/app/api/peanut/user/get-user-from-cookie/route.ts]: nice

File: src/app/api/peanut/user/get-user-from-cookie/route.ts [hunk 5]
```diff
@@ -1,16 +1,15 @@
...
 
         const response = await fetch(apiUrl, {
             method: 'POST',
```
Comment (Hugo0) [inline src/app/api/peanut/user/get-user-from-cookie/route.ts:15]: I'm confused, in our FE we're adding authToken in body, but in our API `verifyAuth` and `requireApiKey` should both fail for this request?

File: src/components/Claim/Link/Onchain/Confirm.view.tsx [hunk 6]
```diff
@@ -253,7 +253,7 @@ export const ConfirmClaimLinkView = ({
                     )}
                 </button>
                 <button className="btn btn-xl dark:border-white dark:text-white" onClick={onPrev} disabled={isLoading}>
-                    Return
+                    Go Back
```
Comment (Hugo0) [inline src/components/Claim/Link/Onchain/Confirm.view.tsx:256]: thx

File: src/app/api/peanut/user/get-user-from-cookie/route.ts (hunk 5 above)
Comment (Hugo0) [reply src/app/api/peanut/user/get-user-from-cookie/route.ts:15]: Also, we have get-user and get-user-from-cookie routes - what's the difference?

In general i think  we'd benefit from cleaning up the plethora of routes we seem to have on the client backend

**Review by Hugo0 — COMMENTED** [review]
Some blocking Qs

File: src/app/api/peanut/user/get-user-from-cookie/route.ts (hunk 5 above)
Comment (jjramirezn) [reply src/app/api/peanut/user/get-user-from-cookie/route.ts:15]: `get-user` seems like it's not being used anywhere

**Review by Hugo0 — APPROVED** [review]

**Pull request merged**